src/gpio/convert.rs

```diff
@@ -0,0 +1,372 @@
+use super::*;
+
+/// Const assert hack
+struct Assert<const L: u8, const R: u8>;
+
+impl<const L: u8, const R: u8> Assert<L, R> {
+    pub const LESS: u8 = R - L - 1;
+}
+
+impl<MODE, HL, const P: char, const N: u8> Pin<MODE, HL, P, N> {
+    fn set_alternate<const A: u8>(&mut self) {
+        #[allow(path_statements, clippy::no_effect)]
+        {
+            Assert::<A, 16>::LESS;
+        }
+        let offset = 2 * { N };
+        unsafe {
+            if N < 8 {
+                let offset2 = 4 * { N };
+                (*Gpio::<P>::ptr()).afrl.modify(|r, w| {
+                    w.bits((r.bits() & !(0b1111 << offset2)) | ((A as u32) << offset2))
+                });
+            } else {
+                let offset2 = 4 * { N - 8 };
+                (*Gpio::<P>::ptr()).afrh.modify(|r, w| {
+                    w.bits((r.bits() & !(0b1111 << offset2)) | ((A as u32) << offset2))
+                });
+            }
+            (*Gpio::<P>::ptr())
+                .moder
+                .modify(|r, w| w.bits((r.bits() & !(0b11 << offset)) | (0b10 << offset)));
+        }
+    }
+    /// Configures the pin to operate alternate mode
+    pub fn into_alternate<const A: u8>(
```

Comment:
The corresponding open-drain method is called `into_alternate_open_drain` so I find `into_alternate_push_pull` more consistent here.

Reply:
It is `PushPull` just in terms of configuration register. Physically it is often even not an Output (depends from peripheral that uses pin).

Reply:
I know but at some point, the user will see that the peripheral requires a `Pin<Alternate<PushPull, ...>, ...>`. Then, it will still be helpful to know that this is the right method to call.